apps/src/dropletUtils.js

```diff
@@ -158,52 +158,44 @@ standardConfig.categories = {
  * @returns {Array}
  */
 function mergeFunctionsWithConfig(codeFunctions, dropletConfig, otherConfig) {
+  if (!codeFunctions || !dropletConfig || !dropletConfig.blocks) {
+    return [];
+  }
+
   var merged = [];
 
-  if (codeFunctions && dropletConfig && dropletConfig.blocks) {
-    var blockSets = [ dropletConfig.blocks ];
-    if (otherConfig) {
-      blockSets.splice(0, 0, otherConfig.blocks);
-    }
-    // codeFunctions is an object with named key/value pairs
-    //  key is a block name from dropletBlocks or standardBlocks
-    //  value is an object that can be used to override block defaults
-    for (var s = 0; s < blockSets.length; s++) {
-      var blocks = blockSets[s];
-      for (var i = 0; i < blocks.length; i++) {
-        var block = blocks[i];
-        if (blocks[i].func in codeFunctions) {
-          // We found this particular block, now override the defaults with extend
-          merged.push(utils.extend(blocks[i], codeFunctions[blocks[i].func]));
-        }
+  var blockSets = [ dropletConfig.blocks ];
+  if (otherConfig) {
+    blockSets.splice(0, 0, otherConfig.blocks);
+  }
+
+  // codeFunctions is an object with named key/value pairs
+  //  key is a block name from dropletBlocks or standardBlocks
+  //  value is an object that can be used to override block defaults
+  for (var s = 0; s < blockSets.length; s++) {
+    var set = blockSets[s];
+    for (var i = 0; i < set.length; i++) {
+      var block = set[i];
+      if (block.func in codeFunctions) {
+        // We found this particular block, now override the defaults with extend
+        merged.push($.extend({}, block, codeFunctions[block.func]));
       }
     }
   }
+
```

Comment:
This should be pretty much the same as before. I just removed a level of nesting, and then renamed some variables (I find it hard to distinguish things like blockSet vs. blockSets). Also, we previously did `var block = blocks[i];` but then proceeded to use `blocks[i]` everywhere instead of `block`.